pkg/fqdn/helpers.go

```diff
@@ -91,7 +93,7 @@ func injectToCIDRSetRules(rule *api.Rule, cache *DNSCache, reMap *regexpmap.Rege
 				lookupIPs := cache.LookupByRegexp(patternRE)
 
 				// Mark this pattern missing; it will be unmarked in the loop below
-				missing[ToFQDN.MatchPattern] = struct{}{}
+				missing[ToFQDN] = struct{}{}
```

Comment:
I think it's fine because we don't usually have matchName and matchPattern set in the same rule, but because we are using the whole selector we have to check that we didn't find IPs for the matchName in this selector and vice versa. :/

Reply:
Do you think we should require that MatchName and MatchPattern be a OneOf? We could always split them up into two separate `FQDNSelectors` in the repository for easier management? :/